Networking/WebSockets/WebSocketProtocol.hh

```diff
@@ -332,23 +333,29 @@ namespace uWS {
             memcpy(dst + headerLength, src, length);
 
             if ( !isServer ) {
-                uint32_t mask_i32 = *reinterpret_cast<uint32_t*>(mask);
+                std::byte* start_byte = dst + headerLength;
+                std::byte* end_byte   = start_byte + length;
+                size_t     i          = 0;
 
-                auto*     start = reinterpret_cast<uint32_t*>(dst + headerLength);
-                uint32_t* end   = start + length / 4;
+                // Handle first x amount of bytes individually until we are aligned with the 4-byte alignment
+                for ( ; reinterpret_cast<uintptr_t>(start_byte) % 4 != 0 && start_byte != end_byte; ++i ) {
+                    *start_byte++ ^= mask[i % 4];
+                }
 
+                // Process majority of bytes in chunks of 4
+                auto*     start = reinterpret_cast<uint32_t*>(start_byte);
+                uint32_t* end   = start + ((end_byte - start_byte) / 4);
+                // Construct mask_i32 such that it's a rotation of 'mask' following on from the first loop
+                uint32_t mask_i32 = 0;
+                for ( int j = 0; j < 4; ++j ) { mask_i32 |= (static_cast<uint32_t>(mask[(i + j) % 4]) << (8 * j)); }
```

Comment:
This concerns me slightly because of the assumed endianness here, but if the tests pass that is probably good enough since I think all of our platforms have the same endianness.  Was this a consideration here?

Reply:
Something I considered, but have no idea how to handle, so I'm glad you brought it up. Is there a good standard way we have to handle cases like this?